File: NewsDemo/app/src/test/java/com/kevin/newsdemo/user/model/api/MockInterceptor.java
package com.kevin.newsdemo.user.model.api;

import okhttp3.*;

import java.io.IOException;

/**
 * Created by kevin on 2019/07/20 22:41.
 */
public class MockInterceptor implements Interceptor {
    private final int code;
    private final String responseString;

    public MockInterceptor(String responseString) {
        this(200, responseString);
    }

    public MockInterceptor(int code, String responseString) {
        this.code = code;
        this.responseString = responseString;
    }

    @Override
    public Response intercept(Interceptor.Chain chain) throws IOException {

        Response response = new Response.Builder()
          .code(code)
          .message(responseString)
          .request(chain.request())
          .protocol(Protocol.HTTP_1_0)
          .body(ResponseBody.create(MediaType.parse("application/json"), responseString.getBytes()))
          .addHeader("content-type", "application/json")
          .build();

        chain.proceed(chain.request());
        return response;
    }
}

File: NewsDemo/app/src/test/java/com/kevin/newsdemo/user/model/api/MockInterceptor.kt
package com.kevin.newsdemo.user.model.api

import okhttp3.Interceptor
import okhttp3.MediaType.Companion.toMediaTypeOrNull
import okhttp3.Protocol
import okhttp3.Response
import okhttp3.ResponseBody
import java.io.IOException

/**
 * Created by kevin on 2019/07/20 22:41.
 */
class MockInterceptor(private val code: Int, private val responseString: String) : Interceptor {

    constructor(responseString: String) : this(200, responseString) {}

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {

        val response = Response.Builder()
                .code(code)
                .message(responseString)
                .request(chain.request())
                .protocol(Protocol.HTTP_1_0)
                .body(ResponseBody.create("application/json".toMediaTypeOrNull(), responseString.toByteArray()))
                .addHeader("content-type", "application/json")
                .build()

        chain.proceed(chain.request())
        return response
    }
}
